consensus/istanbul/backend/engine.go

```diff
@@ -560,10 +561,53 @@ func (sb *Backend) SetChain(chain consensus.ChainReader, currentBlock func() *ty
 	sb.chain = chain
 	sb.currentBlock = currentBlock
 	sb.stateAt = stateAt
+
+	if bc, ok := chain.(*ethCore.BlockChain); ok {
+		go sb.newChainHeadLoop(bc)
+		go sb.updateReplicaStateLoop(bc)
+	}
+
 }
 
-// StartValidating implements consensus.Istanbul.StartValidating
-func (sb *Backend) StartValidating(hasBadBlock func(common.Hash) bool,
+func (sb *Backend) newChainHeadLoop(bc *ethCore.BlockChain) {
+	// Batched. For stats & announce
+	chainHeadCh := make(chan ethCore.ChainHeadEvent, 10)
+	chainHeadSub := bc.SubscribeChainHeadEvent(chainHeadCh)
+	defer chainHeadSub.Unsubscribe()
+
+	for {
+		select {
+		case chainHeadEvent := <-chainHeadCh:
+			sb.newChainHead(chainHeadEvent.Block)
+		case err := <-chainHeadSub.Err():
+			log.Error("Error in istanbul's subscription to the blockchain's chainhead event", "err", err)
```

Comment:
same here... newChainHead() doesn't really communicate much... metrics it's ok to skip, but current valSet looks crucial...

if we are going to miss everything we should "panic" (cleanly)